### PR TITLE
Add method for updating streams files at runtime

### PR DESCRIPTION
**Pull request by xylar**
This is similar to the approach for updating namelist files, and can be used when config options that users might edit could affect things like output interval.

### CODE REVIEW

**Comment by xylar** [comment]
## Testing

I have tested this as part of #238

**Comment by xylar** [comment]
@matthewhoffman, could you review this by inspection?  Since it affects the framework, I just want another pair of eyes on it.

**Review by matthewhoffman — COMMENTED** [review]
@xylar, this makes sense.  Do you think there should be a message in the log file indicating the replacement that was made?  I could imagine a user being confused if they made their own changes to the streams file that then got modified during compass.run.

**Comment by xylar** [comment]
@matthewhoffman, I agree that's a good suggestion (and the same thing for namelist options that are updated at runtime).  I'll add both changes to this PR before I merge.

**Comment by xylar** [comment]
I retested this with #238 and with the `pr` test suite, and it worked as expected.  The planar convergence test gave the expected warnings about streams files being updated at runtime:
```
Warning: updating streams in streams.ocean using the following template and replacements:
compass.ocean.tests.planar_convergence streams.template
output_interval = 001_00:00:00
```

**Comment by xylar** [comment]
@matthewhoffman, when you are ready to do so, could you approve this PR?  I need it to be able to merge #238.

**Review by matthewhoffman — APPROVED** [review]
@xylar , I'm approving the additional changes by inspection, because I don't want to take the time to set up a test that would use the new functionality.  Thank you for pasting the output snippet of these changes in use - that looks good to me.

**Comment by xylar** [comment]
Yep, that's totally fair.  Thank you for the review!

**Pull request merged**
